src/Microsoft.SqlTools.ServiceLayer/AzureBlob/BlobService.cs

```diff
@@ -0,0 +1,78 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.AzureBlob.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+
+namespace Microsoft.SqlTools.ServiceLayer.AzureBlob
+{
+    public class BlobService
+    {
+        private static readonly Lazy<BlobService> instance = new Lazy<BlobService>(() => new BlobService());
+
+        /// <summary>
+        /// Default, parameterless constructor.
+        /// </summary>
+        internal BlobService()
+        {
+        }
+
+        /// <summary>
+        /// Gets the singleton instance object
+        /// </summary>
+        public static BlobService Instance
+        {
+            get { return instance.Value; }
+        }
+
+        public void InitializeService(IProtocolEndpoint serviceHost)
+        {
+            serviceHost.SetRequestHandler(CreateSasRequest.Type, HandleCreateSasRequest);
+        }
+
+        internal async Task HandleCreateSasRequest(
+           CreateSasParams optionsParams,
+           RequestContext<CreateSasResponse> requestContext)
+        {
+            try
+            {
+                ConnectionInfo connInfo;
+                ConnectionService.Instance.TryFindConnection(
+                       optionsParams.OwnerUri,
+                       out connInfo);
+                var response = new CreateSasResponse();
+
+                if (connInfo != null && !connInfo.IsCloud)
+                {
+                    using (
+                    SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, "AzureBlob"))
+                    {
+                        // Connection gets discounnected when backup is done
+                        ServerConnection serverConnection = new ServerConnection(sqlConn);
+                        Server sqlServer = new Server(serverConnection);
+
+                        SharedAccessSignatureCreator sharedAccessSignatureCreator = new SharedAccessSignatureCreator(sqlServer);
+                        string sharedAccessSignature = sharedAccessSignatureCreator.CreateSqlSASCredential(optionsParams.StorageAccountName, optionsParams.BlobContainerKey, optionsParams.BlobContainerUri, optionsParams.ExpirationDate);
+                        response.SharedAccessSignature = sharedAccessSignature;
+                    }
+                } else
+                {
+                    await requestContext.SendError("Create shared access signature is not supported operation.");
```

Comment:
1. You should localize this string since we're likely going to want to show this to the user
2. I'd suggest making it clearer why it failed. There's two cases to consider, either we couldn't find the connection or it isn't a cloud DB. Ideally I'd suggest sending separate messages for both those cases
3. You should return after this, otherwise you'll also end up sending a result on line 70 which is invalid (you should also probably move the SendResult to line 65 since it doesn't make sense to send a result any other time. But it'd still be good to return early)